lib/rules/property-no-deprecated/index.mjs

```diff
@@ -0,0 +1,123 @@
+import { isRegExp, isString } from '../../utils/validateTypes.mjs';
+import isStandardSyntaxDeclaration from '../../utils/isStandardSyntaxDeclaration.mjs';
+import optionsMatches from '../../utils/optionsMatches.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'property-no-deprecated';
+
+const messages = ruleMessages(ruleName, {
+	expected: (unfixed, fixed) => `Expected "${unfixed}" to be "${fixed}"`,
+	rejected: (property) => `Unexpected deprecated property "${property}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/property-no-deprecated',
+	fixable: true,
+};
+
+/** @type {Record<string, string | null>} */
+const DEPRECATED_PROPS_REMAP = {
+	'-khtml-box-direction': null,
+	'-khtml-box-flex': 'flex-grow',
+	'-khtml-box-ordinal-group': 'order',
+	'-khtml-box-pack': null,
+	'-khtml-user-modify': null,
+	'-moz-box-direction': null,
+	'-moz-box-flex': 'flex-grow',
+	'-moz-box-ordinal-group': 'order',
+	'-moz-box-pack': null,
+	'-moz-user-modify': null,
+	'-webkit-box-align': 'align-items',
+	'-webkit-box-direction': null,
+	'-webkit-box-flex': 'flex-grow',
+	'-webkit-box-ordinal-group': 'order',
+	'-webkit-box-orient': null,
+	'-webkit-box-pack': null,
+	'-webkit-user-modify': null,
+	'grid-column-gap': 'column-gap',
+	'grid-gap': 'gap',
+	'grid-row-gap': 'row-gap',
+	'image-orientation': null,
```

Comment:
major

```suggestion
```

> [Values other than `none` and `from-image` are optional to implement and deprecated in CSS.](https://github.com/mdn/content/issues/6820#issuecomment-890864060)

i.e. it's already covered by
https://github.com/stylelint/stylelint/blob/6b505827968f1017b1506ca3d3a7ae526fd4131e/lib/rules/declaration-property-value-keyword-no-deprecated/index.mjs#L97